functions-python/feed_sync_process_transitland/src/main.py

```diff
@@ -0,0 +1,381 @@
+            raise
+
+    def process_new_feed(self, payload: FeedPayload) -> None:
+        """
+        Process creation of a new feed
+
+        Args:
+            payload (FeedPayload): The feed payload for new feed
+        """
+
+        logger.info(
+            f"Starting new feed creation for external_id: {payload.external_id}"
+        )
+
+        # Check if feed with same URL exists
+        if self.check_feed_url_exists(payload.feed_url):
+            logger.info(f"Feed URL already exists: {payload.feed_url}")
+            return
+
+        # Generate new feed ID and stable ID
+        feed_id = str(uuid.uuid4())
+        stable_id = f"{payload.source}-{payload.external_id}"
+
+        logger.debug(f"Generated new feed_id: {feed_id} and stable_id: {stable_id}")
+
+        try:
+            # Create new feed
+            new_feed = Feed(
+                id=feed_id,
+                data_type=payload.spec,
+                feed_name=f"Feed from {payload.operator_name}"
+                if payload.operator_name
+                else "Unnamed Feed",
+                producer_url=payload.feed_url,
+                authentication_type=payload.type if payload.type else "0",
+                authentication_info_url=payload.auth_info_url,
+                api_key_parameter_name=payload.auth_param_name,
+                stable_id=stable_id,
+                status="active",
+                provider=payload.operator_name,
+            )
+
+            # external ID mapping
+            external_id = Externalid(
+                feed_id=feed_id,
+                associated_id=payload.external_id,
+                source=payload.source,
+            )
+
+            # Add relationships
+            new_feed.externalids.append(external_id)
+
+            self.session.add(new_feed)
+            self.session.flush()
+
+            logger.debug(f"Successfully created feed with ID: {feed_id}")
+            logger.info(
+                f"Created new feed with ID: {feed_id} for external_id: {payload.external_id}"
+            )
+
+        except Exception as e:
+            logger.error(
+                f"Error creating new feed for external_id {payload.external_id}: {str(e)}"
+            )
+            raise
+
+    def process_feed_update(self, payload: FeedPayload, old_feed_id: str) -> None:
+        """
+        Process feed update when URL has changed
+
+        Args:
+            payload (FeedPayload): The feed payload for update
+            old_feed_id (str): The ID of the existing feed to be updated
+        """
+
+        logger.info(
+            f"Starting feed update process for external_id: {payload.external_id}"
+        )
+        logger.debug(f"Old feed_id: {old_feed_id}, New URL: {payload.feed_url}")
+
+        try:
+            # Create new feed with updated URL
+            new_feed_id = str(uuid.uuid4())
+            stable_id = f"{payload.source}-{payload.external_id}"
```

Comment:
If I'm not mistaking, following the code logic, the updated feed would have the same `stable_id` as the old version. Is this behaviour ok as we use the `stable_id` as a key in a lot of processes including the feed page url in the web ui? @emmambd @davidgamez

Reply:
See above - is this in cases where the URLs match? If so, there should be no new feed

Reply:
This is the same case as the previous explanation